File: 01_Language/01_Functions/php/date_parse_from_format.php
<?php

$date = DateTime::createFromFormat('j-M-Y', '15-Feb-2009');
echo $date->format('Y-m-d');

$date = date_create_from_format('j-M-Y', '15-Feb-2009');
echo date_format($date, 'Y-m-d');

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/DateParseFromFormat.java
package com.onevgo.functions;

import org.apache.commons.lang3.time.DateUtils;

import java.text.ParseException;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Date;
import java.util.Locale;

public class DateParseFromFormat {
    public static void main(String[] args) throws ParseException {
        Date dateTime = DateUtils.parseDate("15-Feb-2009", Locale.ENGLISH, "dd-MMM-yyyy");
        System.out.println(dateTime);

        //jdk8+
        System.out.println(LocalDate.parse("15-Feb-2009", DateTimeFormatter.ofPattern("dd-MMM-yyyy", Locale.ENGLISH)));
    }
}
